### ISSUE TITLE
--val_best during train based on val_loss or psuedo-dice?

### ISSUE THREAD

**Issue by rbramkumar** [MIC-DKFZ/nnUNet#2754]
I am new to this implementation so I'm wondering whether the "--val_best" option does what I think it does. In my use-case, it looks like the val loss stops decreasing quite early. Do you know if the checkpoint model saves the best model according to "val_loss" (red line) or "pseudo-dice" (green line). Is there a way to save the best model according to the evaluation metric i.e., green line? 

Also, putting it out there in case anyone knows more than me here - any suggestions to bring down val_loss? 

`!nnUNetv2_train 555 2d 0  -p nnUNetResEncUNetMPlans --val_best`

![Image](https://github.com/user-attachments/assets/4ae30d9e-7e6f-4c63-ac69-902edc76e58f)

**Comment by constantinulrich** [comment]
Hi, we calculate an exponential moving average on the pseudo dice for the "best" checkpoint. But take that with care. We do not do a proper validation - its just a pseudo dice with 50 foreground patches.

**Comment by constantinulrich** [comment]
> Also, putting it out there in case anyone knows more than me here - any suggestions to bring down val_loss?

There are trainers implemented that use a stronger data augmentation scheme (nnUNetTrainerDA5), which you could test...

**Comment by rbramkumar** [comment]
Thank you very much. I will try out the trainer with data augmentations. 

Thanks,
Ram
